pkg/test/framework/components/apps/kube.go

```diff
@@ -641,21 +641,25 @@ func (a *kubeApp) Call(e AppEndpoint, opts AppCallOptions) ([]*echo.ParsedRespon
 	dstURL := dst.makeURL(opts)
 	dstServiceName := dst.owner.Name()
 
-	// If host header is set, override the destination with it
-	if opts.Headers.Get("Host") != "" {
-		dstServiceName = opts.Headers.Get("Host")
+	protoHeaders := []*proto.Header{
+		{
+			Key:   "Host",
+			Value: dstServiceName,
+		},
+	}
+
+	// Add headers in opts.Headers, e.g., authorization header, etc.
+	// If host header is set, it will override dstServiceName
+	for k := range opts.Headers {
```

Comment:
Please add this same change to the kube version of echo (I'm about replace the apps component with echo).

Reply:
Done.